From d94da3ab78b37333d259a9cb5d1f0008f7f214f7 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Mon, 29 Jan 2024 12:11:09 -0800
Subject: [PATCH 5/7] rocm 6.0.0 CUDAHooks.cpp fix

CUDAHooks.cpp is used to generate HIPHooks.cpp
and hipPointerAttribute_t class memoryType
was renamed to type on rocm 6.0.0 version.

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 aten/src/ATen/cuda/detail/CUDAHooks.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/aten/src/ATen/cuda/detail/CUDAHooks.cpp b/aten/src/ATen/cuda/detail/CUDAHooks.cpp
index acb9b1931f0..8d1fa94a630 100644
--- a/aten/src/ATen/cuda/detail/CUDAHooks.cpp
+++ b/aten/src/ATen/cuda/detail/CUDAHooks.cpp
@@ -154,7 +154,11 @@ bool CUDAHooks::isPinnedPtr(const void* data) const {
 #if !defined(USE_ROCM)
   return attr.type == cudaMemoryTypeHost;
 #else
-  return attr.memoryType == cudaMemoryTypeHost;
+  #if ROCM_VERSION >= 60000
+    return attr.type == hipMemoryTypeHost;
+  #else
+    return attr.memoryType == hipMemoryTypeHost;
+  #endif
 #endif
 }
 
-- 
2.41.0

